.github/copilot-instructions.md

```diff
@@ -0,0 +1,197 @@
+A change is considered WASM/WASI-relevant if:
+- The relevant `.csproj` contains explicit Browser/WASM or WASI targets (look for `<TargetFrameworks>`, `$(TargetPlatformIdentifier)`, or `Condition` attributes referencing `browser` or `wasi`, as well as `TARGET_BROWSER` or `TARGET_WASI` constants), **and**
+- The changed file is not excluded from the build for that platform in any way with a `Condition` attribute on `<ItemGroup>` or `<Compile>`.
+
+### 1.2. Baseline Setup
+
+Ensure you have a full successful build of the needed runtime+libraries as a baseline.
+
+1. Checkout `main` branch
+
+2. From the repository root, run the build depending on the affected component. If multiple components are affected, subsequently run and verify the builds for all of them.
+   - **CoreCLR (CLR):** `./build.sh clr+libs`
+   - **Mono Runtime:** `./build.sh mono+libs`
+   - **Libraries:** `./build.sh clr+libs -rc release`
+   - **WASM/WASI Libraries:** `./build.sh mono+libs -os browser`
+
+3. Verify the build completed without error.
+   - _If the build failed, report the failure and don't proceed with the changes._
+4. From the repository root:
+   - Configure PATH: `export PATH="$(pwd)/.dotnet:$PATH"`
+   - Verify SDK Version: `dotnet --version` should match `sdk.version` in `global.json`.
+5. Switch back to the working branch.
+
+---
+
+## 2. Iterative Build and Test Strategy
+
+1. Apply the intended changes
+2. **Attempt Build.** If the build fails, attempt to fix and retry the step (up to 5 attempts).
+3. **Attempt Test.**
+   * If a test _build_ fails, attempt to fix and retry the step (up to 5 attempts).
+   * If a test _run_ fails,
+      - Determine if the problem is in the test or in the source
+      - If the problem is in the test, attempt to fix and retry the step (up to 5 attempts).
+      - If the problem is in the source, reconsider the full changeset, attempt to fix and repeat the workflow.
+4. **Workflow Iteration:**
+   * Repeat build and test up to 5 cycles.
+   * If issues persist after 5 workflow cycles, report failure.
+   * If the same error persists after each fix attempt, do not repeat the same fix. Instead, escalate or report with full logs.
+
+When retrying, attempt different fixes and adjust based on the build/test results.
+
+### Success Criteria
+
+* **Build:**
+   - Completes without errors.
+   - Any non-zero exit code from build commands is considered a failure.
+* **Tests:**
+   - All tests must pass (zero failures).
+   - Any non-zero exit code from test commands is considered a failure.
+
+* **On success:** Report completion;
+  - Otherwise, report error(s) with logs for diagnostics.
+    - Collect logs from `artifacts/log/` and the console output for both build and test steps.
+    - Attach relevant log files or error snippets when reporting failures.
+
+---
+
+## 3. CoreCLR (CLR) Workflow
+
+From the repository root:
+
+- Build: `./build.sh -subset clr`
+- Run tests: `cd src/tests && ./build.sh`
```

Comment:
This builds the tests. You need to run `run.sh` to run them. It's mentioned at the bottom of the link below.

Reply:
Thanks